implementations/rust/ockam/ockam_api/src/minicbor_url.rs

```diff
@@ -0,0 +1,16 @@
+use minicbor::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use url::Url as RegularUrl;
+
+#[derive(Encode, Decode, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct Url(#[n(0)] RegularUrl);
```

Comment:
I've implemented newtype here, and added all macros suggested by the compiler to minimize the number of errors.
Now when I run `cargo check` I am getting a similar error to the previous one I've addressed.

'''
error[E0277]: the trait bound `reqwest::Url: minicbor::Encode<Ctx>` is not satisfied
 --> implementations/rust/ockam/ockam_api/src/minicbor_url.rs:5:10
  |
5 | #[derive(Encode, Decode, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
  |          ^^^^^^ the trait `minicbor::Encode<Ctx>` is not implemented for `reqwest::Url`
  |
  = help: the following other types implement trait `minicbor::Encode<C>`:
            <&T as minicbor::Encode<C>>
            <&mut T as minicbor::Encode<C>>
            <() as minicbor::Encode<C>>
            <(A, B) as minicbor::Encode<Ctx>>
            <(A, B, C) as minicbor::Encode<Ctx>>
            <(A, B, C, D) as minicbor::Encode<Ctx>>
            <(A, B, C, D, E) as minicbor::Encode<Ctx>>
            <(A, B, C, D, E, F) as minicbor::Encode<Ctx>>
          and 234 others
  = note: this error originates in the derive macro `Encode` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `reqwest::Url: minicbor::Decode<'_, Ctx>` is not satisfied
 --> implementations/rust/ockam/ockam_api/src/minicbor_url.rs:5:18
  |
5 | #[derive(Encode, Decode, Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
  |                  ^^^^^^ the trait `minicbor::Decode<'_, Ctx>` is not implemented for `reqwest::Url`
  |
  = help: the following other types implement trait `minicbor::Decode<'b, C>`:
            <&'a ByteSlice as minicbor::Decode<'b, C>>
            <&'a std::path::Path as minicbor::Decode<'b, C>>
            <&'a str as minicbor::Decode<'b, C>>
            <() as minicbor::Decode<'b, C>>
            <(A, B) as minicbor::Decode<'b, Ctx>>
            <(A, B, C) as minicbor::Decode<'b, Ctx>>
            <(A, B, C, D) as minicbor::Decode<'b, Ctx>>
            <(A, B, C, D, E) as minicbor::Decode<'b, Ctx>>
          and 228 others
  = note: this error originates in the derive macro `Decode` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ockam_api` due to 2 previous errors
'''

I am not sure why the compiler is referencing `reqwest::Url` since I am using `url::Url` but they seems to be the same. I am a bit confused by this. @etorreborre